### PR TITLE
fix flickering visualization

### PR DESCRIPTION
**Pull request by VladyslavHrynchak200204**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Primary OS tested on | (Ubuntu) |
| Robotic platform tested on | gazebo simulation |

---

**Purpose:**

The map_vis_z parameter has been introduced to help modify the map slightly below the default plane, aiming to eliminate flickering issues.

**Default Value:**

map_vis_z: 0.0

**Minimum Value Without Flickering:**

map_vis_z: -0.008

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by VladyslavHrynchak200204** [comment]
still in progress

**Comment by SteveMacenski** [comment]
OK, I'm going to refrain from a detailed review until you're ready!

**Comment by VladyslavHrynchak200204** [comment]
> OK, I'm going to refrain from a detailed review until you're ready! I am curious however what value you set this to

We tested different value which is as close to 0 as possible without flickering. 

> OK, I'm going to refrain from a detailed review until you're ready! I am curious however what value you set this to

We tested different value which is as close to 0 as possible without flickering. From out testing map_vis_z: -0.008 is minimum value for this purpose.

**Comment by VladyslavHrynchak200204** [comment]
> OK, I'm going to refrain from a detailed review until you're ready! I am curious however what value you set this to

We tested different value which is as close to 0 as possible without flickering. From out testing map_vis_z: -0.008 is minimum value for this purpose.

**Comment by VladyslavHrynchak200204** [comment]
> > OK, I'm going to refrain from a detailed review until you're ready! I am curious however what value you set this to
> 
> We tested different value which is as close to 0 as possible without flickering. From out testing map_vis_z: -0.008 is minimum value for this purpose.

@SteveMacenski  Could you check these changes please?

File: nav2_costmap_2d/src/costmap_2d_publisher.cpp [hunk 1]
```diff
@@ -54,12 +54,14 @@ Costmap2DPublisher::Costmap2DPublisher(
   Costmap2D * costmap,
   std::string global_frame,
   std::string topic_name,
-  bool always_send_full_costmap)
+  bool always_send_full_costmap,
+  float map_vis_z)
```
Comment (SteveMacenski) [inline nav2_costmap_2d/src/costmap_2d_publisher.cpp:58]: double

File: nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp [hunk 2]
```diff
@@ -404,6 +404,7 @@ class Costmap2DROS : public nav2_util::LifecycleNode
   bool track_unknown_space_{false};
   double transform_tolerance_{0};           ///< The timeout before transform errors
   double initial_transform_timeout_{0};   ///< The timeout before activation of the node errors
+  float map_vis_z_{0};
```
Comment (SteveMacenski) [inline nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp:407]: double

File: nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_publisher.hpp [hunk 3]
```diff
@@ -166,6 +167,7 @@ class Costmap2DPublisher
   double saved_origin_y_;
   bool active_;
   bool always_send_full_costmap_;
+  float map_vis_z_;
```
Comment (SteveMacenski) [inline nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_publisher.hpp:170]: double

File: nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_publisher.hpp [hunk 4]
```diff
@@ -72,7 +72,8 @@ class Costmap2DPublisher
     Costmap2D * costmap,
     std::string global_frame,
     std::string topic_name,
-    bool always_send_full_costmap = false);
+    bool always_send_full_costmap = false,
+    float map_vis_z = 0.0);
```
Comment (SteveMacenski) [inline nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_publisher.hpp:76]: double

**Review by SteveMacenski — COMMENTED** [review]
The parameter / value in the occ grid are doubles, so keep them doubles please. Otherwise LGTM. 

You do need to add to:
- the parameter guide the new param
- the migration guide the new param and reasons - an image would be nice

**Comment by VladyslavHrynchak200204** [comment]
> The parameter / value in the occ grid are doubles, so keep them doubles please. Otherwise LGTM.
> 
> You do need to add to:
> 
> * the parameter guide the new param
> * the migration guide the new param and reasons - an image would be nice

Done. https://github.com/ros-navigation/docs.nav2.org/pull/578

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
CI failed in a weird way, so I retriggered that - so waiting to merge the software pending that. Also, your docs are missing the configuration guide update - please update that and I can merge the pair! Thanks!

**Comment by VladyslavHrynchak200204** [comment]
> CI failed in a weird way, so I retriggered that - so waiting to merge the software pending that. Also, your docs are missing the configuration guide update - please update that and I can merge the pair! Thanks!

Done

**Comment by VladyslavHrynchak200204** [comment]
I pulled the last changes and test have passed.

**Comment by VladyslavHrynchak200204** [comment]
> I pulled the last changes and test have passed. Does codecov need to be fixed as well?

**Comment by SteveMacenski** [comment]
No, that is fine, thanks for the follow up!

**Pull request merged**